### PR TITLE
seqlink 1.0.4

### PR DESCRIPTION
**Pull request by changebio**
1. Implement get AF from annotation.
2. Fix maf bug by [reverse genotypes if maf > 0.5](https://github.com/gaow/SEQLinkage/commit/93caaa36682c1bd806ca473f96f73964700bf420)
3. Fix output format and path bugs.
4. Update the command line script.

### CODE REVIEW

File: SEQLinkage/Main.py [hunk 1]
```diff
@@ -28,16 +28,16 @@ class Args:
...
-        prog = env.prog,
+        prog = 'seqlink',
         fromfile_prefix_chars = '@', add_help = False,
         epilog = '''\tCopyright (c) 2013 - 2014 Gao Wang <wang.gao@columbia.edu>\n\tDistributed under GNU General Public License\n\tHome page: {}'''.format(HOMEPAGE))
         self.getEncoderArguments(self.parser)
         self.getIOArguments(self.parser)
         self.getLinkageArguments(self.parser)
         self.getRuntimeArguments(self.parser)
-        self.parser.set_defaults(func=main)
+        #self.parser.set_defaults(func=main)
```
Comment (gaow) [inline SEQLinkage/Main.py:40]: I cannot remember what this does, but it seems necessary to me? If it is truly unnecessary you should remove it and not comment it out?

File: SEQLinkage/Main.py [hunk 2]
```diff
@@ -152,9 +149,11 @@ def checkParams(args):
     return True
 
 # Cell
-def main(args):
+def main():
     '''the main encoder function'''
+    args = Args().get()
     checkParams(args)
+    print(env.version,'env')
```
Comment (gaow) [inline SEQLinkage/Main.py:156]: Please don't print anything

File: SEQLinkage/Main.py [hunk 3]
```diff
@@ -260,6 +259,7 @@ def main(args):
     env.jobs = args.jobs
     # STEP 2: write to PLINK or mega2 format
     tpeds = [os.path.join(env.tmp_cache, item) for item in os.listdir(env.tmp_cache) if item.startswith(env.output) and item.endswith('.tped')]
+    print(tpeds) #testing line
```
Comment (gaow) [inline SEQLinkage/Main.py:262]: don't print it here.

File: SEQLinkage/Runner.py [hunk 4]
```diff
@@ -51,7 +53,8 @@ def format_plink(tped, tfam):
         with open(out_base + '.ped', 'w') as p:
             for line in tfam_fh:
                 p.write(line.strip())
-                list(map(lambda x: p.write(' {} {}'.format(x.popleft(), x.popleft)), geno))
+                s = line.strip().split()
```
Comment (gaow) [inline SEQLinkage/Runner.py:56]: do u use this `s` later? if not please remove

File: SEQLinkage/Runner.py [hunk 5]
```diff
@@ -139,7 +142,7 @@ def format_merlin(tped, tfam):
 #because the haplotype patterns are different from family to family.
 #You can analyze them all together
 def format_linkage(tped, tfam, prev, wild_pen, muta_pen, inherit_mode, theta_max, theta_inc):
-    out_base = '{}/LINKAGE/{}'.format(env.tmp_dir, splitext(basename(tped))[0])
+    out_base = '{}/LINKAGE/{}'.format(env.outdir, splitext(basename(tped))[0])
```
Comment (gaow) [inline SEQLinkage/Runner.py:145]: Please double-check this. I think we distinguished between temp and out dir for a reason: we can safely remove anything in temp dir and only keep the out dir at the end of the analysis.

File: SEQLinkage/Runner.py (hunk 4 above)
Comment (changebio) [reply SEQLinkage/Runner.py:56]: Yes. without this line, the format will be wrong.

File: SEQLinkage/Main.py (hunk 2 above)
Comment (changebio) [reply SEQLinkage/Main.py:156]: This line is a test line, which will be removed.

File: SEQLinkage/Main.py (hunk 1 above)
Comment (changebio) [reply SEQLinkage/Main.py:40]: make install easier with pip

File: SEQLinkage/Main.py [hunk 6]
```diff
@@ -65,6 +64,8 @@ def getIOArguments(self, parser):
         vargs.add_argument('--fam', metavar='FILE', required=True, dest = "tfam",
                            help='''Input pedigree and phenotype information in FAM format.''')
         vargs.add_argument('--vcf', metavar='FILE', required=True, help='''Input VCF file, bgzipped.''')
+        vargs.add_argument('--anno', metavar='FILE', required=False, help='''Input annotation file from annovar.''')
+        vargs.add_argument('--pop', metavar='FILE', required=False, help='''Input two columns file, first column is family ID,second column population information.''')
```
Comment (gaow) [inline SEQLinkage/Main.py:68]: extra `,` please remove

Comment (gaow) [inline SEQLinkage/Main.py:68]: there should be a space between `,` and `second`

**Comment by gaow** [comment]
@changebio please pin me when you are done with resolving all my comments (could you prioritize on that).

File: SEQLinkage/Runner.py (hunk 5 above)
Comment (changebio) [reply SEQLinkage/Runner.py:144]: This is the linkage format output.

**Comment by changebio** [comment]
Resolving above comments. @gaow

**Comment by gaow** [comment]
@changebio i merged it. But CI still fails.

**Pull request merged**